src/Stratis.Features.FederatedPeg/TargetChain/CrossChainTransferStore.cs

```diff
@@ -920,6 +920,7 @@ private void SaveNextMatureHeight(DBreeze.Transactions.Transaction dbreezeTransa
         }
 
         /// <inheritdoc />
+        [Obsolete("This doesn't lock when validating cross chain transfers. Don't use.")]
```

Comment:
We avoid using this method (it's only used in tests now) as it called ValidateCrossChainTransfers out of a lock.